extensions/azurecore/src/azureResource/services/subscriptionService.ts

```diff
@@ -3,14 +3,30 @@
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { Account } from 'azdata';
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
 import { SubscriptionClient } from '@azure/arm-subscriptions';
 
 import { azureResource } from 'azureResource';
 import { IAzureResourceSubscriptionService } from '../interfaces';
+import { TokenCredentials } from '@azure/ms-rest-js';
+import { AzureSubscriptionError } from '../errors';
+import { AzureResourceErrorMessageUtil } from '../utils';
+
+import * as nls from 'vscode-nls';
+import { AzureAccount } from 'azurecore';
+const localize = nls.loadMessageBundle();
 
 export class AzureResourceSubscriptionService implements IAzureResourceSubscriptionService {
-	public async getSubscriptions(account: Account, credential: any, tenantId: string): Promise<azureResource.AzureResourceSubscription[]> {
+	/**
+	 * Gets all of the subscriptions for the specified account using the specified credential. This assumes that the credential passed is for
+	 * the specified tenant - which the subscriptions returned will be associated with.
+	 * @param account The account to get the subscriptions for
+	 * @param credential The credential to use for querying the subscriptions
+	 * @param tenantId The ID of the tenant these subscriptions are for
+	 * @returns The list of all subscriptions on this account for the specified tenant
+	 */
+	public async getSubscriptions(account: azdata.Account, credential: any, tenantId: string): Promise<azureResource.AzureResourceSubscription[]> {
```

Comment:
why do we need credential parameter here but not in the new method, it seems to me we can make these 2nd and 3rd parameters optional (if we really need 2nd parameter) without introducing another method.

Reply:
I considered that, but I thought it would be slightly confusing since if you'd pass in one then it would fail (without more changes at least). 

I don't really feel there's a clear best answer here - one extra function isn't a huge deal considering this isn't exported anywhere and it makes sure that people are clear on the specific use for each type.